### PR TITLE
Input set component

### PR DESCRIPTION
**Pull request by dylanaubrey**
Creating an input fieldset component that combines a label, input and a error message. Linked to enhancement https://github.com/dylanaubrey/component-library/issues/27.

### CODE REVIEW

**Comment by dylanaubrey** [comment]
@petewarman can you have a look at this... I thought it would be a good example of a simple composed component... not sure about the best way to pass down properties specific to each sub component so have gone with specific properties for the label and the input, but not really happy with this approach. What do you think?

File: packages/input-fieldset/src/component/styled.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import styled from 'styled-components';
+import { getErrorMessageColor, getMarginBottom } from './getters';
+
+export const StyledErrorMessage = styled.div`
+  color: ${getErrorMessageColor};
+  font-size: 16px;
+  line-height: 1.25;
+`;
+
+export const StyledFieldset = styled.fieldset`
```
Comment (petewarman) [inline packages/input-fieldset/src/component/styled.js]: I'm not sure if this should be a fieldset? I think fieldset is intended to group multiple inputs - eg a group of related radio buttons or a whole section of a form.

File: packages/input-fieldset/src/component/index.jsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      {error && errorMessage && <StyledErrorMessage>{errorMessage}</StyledErrorMessage>}
+    </StyledFieldset>
+  );
+}
+
+InputFieldset.propTypes = {
+  /**
+   * If true, the fieldset will render in an error state.
+   */
+  error: PropTypes.bool,
```
Comment (petewarman) [inline packages/input-fieldset/src/component/index.jsx]: I would suggest we go with something like `isError` or `hasError` here so it's implied by the prop name that it is a boolean value. Makes it less likely people will try to assign it to a string value or and error object or something.

File: packages/input-fieldset/src/component/index.jsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  /**
+   * The message to display in an error state.
+   */
+  errorMessage: PropTypes.string,
```
Comment (petewarman) [inline packages/input-fieldset/src/component/index.jsx]: do we need the other error prop? can we just check if this value is falsey?

File: packages/input-fieldset/src/component/styled.js (hunk 1 above)
Comment (dylanaubrey) [reply packages/input-fieldset/src/component/styled.js]: okay... we can just wrap the label around the input... what about that?

File: packages/input-fieldset/src/component/index.jsx (hunk 3 above)
Comment (dylanaubrey) [reply packages/input-fieldset/src/component/index.jsx]: It depends how it is going to be used right? A user might set up the component with the error message and then just change the state to `error`... or they might just pass down the message if it changes based on the user action...

**Comment by dylanaubrey** [comment]
@petewarman I've updated the PR following our discussions, please have a look... I've been thinking about the error and errorMessage props and I think they should stay as I can think of a few scenarios where you might want to display an error state without a message like when you have grouped fields (I've seen address forms like this).

**Comment by dylanaubrey** [comment]
Fixes #27

**Pull request merged**